src/repo.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * All repo relate configuration lives within this file. Add domains,
+ * effects, and configure any repo startup behavior here.
+ */
+
+import Microcosm from 'microcosm'
+import Game from './domains/game'
+import Timer from './effects/timer'
+
+export default class Repo extends Microcosm {
```

Comment:
`repo.push` dispatches an action with a payload to the domain, which then determines the next state of the game.

So when I toggle a light, it dispatches an action with the x and y position of the light as the payload. The domain then processes that and computes the new state of the whole matrix of lights (such as toggling adjacent lights) and updates the model, which then gets reflected by the `render` statement.